server/tailsql/tailsql_test.go

```diff
@@ -483,3 +483,44 @@ func TestQueryTimeout(t *testing.T) {
 		t.Fatal("Timeout waiting for query to end")
 	}
 }
+
+func TestQueryable(t *testing.T) {
+	_, db := mustInitSQLite(t)
+
+	s, err := tailsql.NewServer(tailsql.Options{
+		Sources: []tailsql.DBSpec{{
+			Source: "quux",
+			Label:  "Happy fun database",
+			DB:     sqlDB{DB: db},
+
+			// Note: Omit Driver to exercise that this is allowed when a
+			// programmatic data source is given.
+		}},
+	})
+	if err != nil {
+		t.Fatalf("NewServer: %v", err)
+	}
+	defer s.Close()
+
+	htest := httptest.NewServer(s.NewMux())
+	defer htest.Close()
+	cli := htest.Client()
+
+	t.Run("Smoke", func(t *testing.T) {
+		const testProbe = "mindlesprocket"
+		q := url.Values{
+			"src": {"quux"},
+			"q":   {fmt.Sprintf(`select '%s'`, testProbe)},
+		}
+		rsp := string(mustGet(t, cli, htest.URL+"/csv?"+q.Encode()))
+		if !strings.Contains(rsp, testProbe) {
+			t.Errorf("Query failed: got %q, want %q", rsp, testProbe)
+		}
+	})
+}
+
+type sqlDB struct{ *sql.DB }
```

Comment:
ah this is a different package `tailscale_test`

Reply:
maybe expose something like `FromSQL`?

Reply:
I was kinda hoping to not need to. Since we have to keep `SetDB` around anyway, the caller I think doesn't need it? (And I don't think a few duplicate lines in the test are worth it)